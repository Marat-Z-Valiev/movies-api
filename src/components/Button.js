import React from "react";
import ButtonStyled from "./styles/ButtonStyled";
import PropTypes from "prop-types";

const Button = ({handleClick, isLoading, isDisabled}) => {
	const handleKeyPress = (event, handleFunction) => {
		if (event.key === "Enter") {
			handleFunction;
		}
	};
	return (
		<ButtonStyled
			type="button"
			onClick={handleClick}
			onKeyPress={event => handleKeyPress(event, handleClick)}
			disabled={isDisabled}
		>
			{isLoading ? "Loading" : "Search"}
		</ButtonStyled>
	);
};

Button.propTypes = {
	handleClick: PropTypes.func,
	isLoading: PropTypes.bool,
	isDisabled: PropTypes.bool
};

export default Button;
