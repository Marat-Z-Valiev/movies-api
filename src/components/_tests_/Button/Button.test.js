import React from "react";
import {shallow, mount, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Button from "../../Button";

configure({adapter: new Adapter()});

describe("Button", () => {
	it("should render correctly", () => {
		const component = shallow(
			<Button handleClick={() => {}} isLoading={false} isDisabled={false} />
		);
		expect(toJSON(component)).toMatchSnapshot();
	});

	it("should trigger the handleClick function on click", () => {
		const handleClick = jest.fn();
		const component = shallow(
			<Button handleClick={handleClick} isLoading={false} isDisabled={false} />
		);
		component.simulate("click");
		expect(handleClick).toHaveBeenCalled();
	});

	it("should have text 'Loading' when isLoading is true", () => {
		const component = mount(
			<Button handleClick={() => {}} isLoading={true} isDisabled={false} />
		);
		expect(component.text()).toBe("Loading");
	});

	it("should be disabled if isDisabled is true", () => {
		const component = shallow(
			<Button handleClick={() => {}} isLoading={false} isDisabled={true} />
		);
		expect(component.props().disabled).toBe(true);
	});
});
