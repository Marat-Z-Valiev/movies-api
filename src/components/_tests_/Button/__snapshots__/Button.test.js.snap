// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<styled.button
  disabled={false}
  onClick={[Function]}
  onKeyPress={[Function]}
  type="button"
>
  Search
</styled.button>
`;
